### PR TITLE
[IMP] fieldservice_stock: Add "Create FSM Equipment" option to Operation Types

### PR DESCRIPTION
**Pull request by dreispt**
FSM Equipment creation for serial number tracked products can be
automated on outgoing/delivey transfers.
This allows for it to happen earlier, on other Operation Types.

### CODE REVIEW

File: fieldservice_stock/models/stock_move.py [hunk 1]
```diff
@@ -20,8 +20,11 @@ def prepare_equipment_values(self, move_line):
     def _action_done(self):
         res = super()._action_done()
         for rec in self:
-            if rec.picking_code == 'outgoing' and rec.state == 'done':
-                if rec.product_tmpl_id.create_fsm_equipment:
+            create_fsm_equipment = rec.product_tmpl_id.create_fsm_equipment
+            if rec.state == 'done' and create_fsm_equipment:
+                is_outgoing = rec.picking_code == 'outgoing'
+                can_create_fsm = rec.picking_type_id.can_create_fsm_equipment
+                if can_create_fsm or is_outgoing:
```
Comment (max3903) [inline fieldservice_stock/models/stock_move.py]: @dreispt 

I think we can simplify this by dropping the picking_code. I can easily see a company  willing to create an equipment when a product is received.

```
if rec.state == 'done' and \
    rec.product_tmpl_id.create_fsm_equipment and \
    rec.picking_type_id.create_fsm_equipment:
```

File: fieldservice_stock/models/stock_picking_type.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright (C) 2020 Open Source Integrators, Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = 'stock.picking.type'
+
+    force_create_fsm_equipment = fields.Boolean(
```
Comment (max3903) [inline fieldservice_stock/models/stock_picking_type.py]:
```suggestion
    create_fsm_equipment = fields.Boolean(
```

File: fieldservice_stock/models/stock_move.py (hunk 1 above)
Comment (dreispt) [reply fieldservice_stock/models/stock_move.py]: I worry about being "stable" and not breaking current installs.
Do you think that is something we could waive?

Comment (max3903) [reply fieldservice_stock/models/stock_move.py]: @dreispt We can have a migration script that check the box on the picking type based on the code.

**Comment by dreispt** [comment]
@max3903 Doing tests, I stumbled upon a required field:
`ERROR: null value in column "current_location_id" violates not-null constraint`
Any suggestion on how to deal with this?

**Comment by dreispt** [comment]
Ready for review.

**Review by osi-scampbell — APPROVED** [review]

**Comment by dreispt** [comment]
@max3903 Good to merge?

**Review by max3903 — APPROVED** [review]

**Pull request merged**